File: 0338-counting-bits/0338-counting-bits.java
class Solution {
    public int[] countBits(int n) {
        int[] dp = new int[n+1];
        int sub = 1;
        for(int i = 1; i <= n; i++){
            dp[i] = dp[i>>1]+(1&i);
        }
        return dp;
    }
}

File: 0338-counting-bits/0338-counting-bits.py
class Solution(object):
    def countBits(self, n):
        dp = []
        dp.append(0)
        sub = 1
        for i in range(1,n+1):
            if sub * 2 == i:
                sub=i
            dp.append(dp[i-sub]+1)
        return dp
        